### ISSUE TITLE
when training with transformer decoder, words can see their right words in self-attention part.

### ISSUE THREAD

**Issue by freefuiiismyname** [asyml/texar#131]
Hi.
I notice that, when training with transformer decoder, words can see their right words in self-attention part.
Is this true?
If yes,I am very confused about that,cuz when inferencing decoding,words dont have right words,that may product problems.So ... I think,maybe mask the right word's logits in self-attention part makes more sense?

**Comment by ZhitingHu** [comment]
it's already masked. decoder can only see w_1 ... w_t-1 when predicting w_t
